carbonmark/components/pages/Projects/Banners/BankTransfer.tsx

```diff
@@ -0,0 +1,108 @@
+import { t } from "@lingui/macro";
+import { ButtonPrimary } from "components/Buttons/ButtonPrimary";
+import { ButtonSecondary } from "components/Buttons/ButtonSecondary";
+import FeatureBanner from "components/FeatureBanner";
+import { urls } from "lib/constants";
+import { FC, useEffect, useState } from "react";
+
+type BannerState = "initial-banner" | "secondary-banner" | "none";
+type SessionState = "visible" | "hidden" | "not-shown";
+
+export const BankTransferBanner: FC = () => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [localStorage, setLocalStorage] =
+    useState<BannerState>("initial-banner");
+  const [sessionStorage, setSessionStorage] = useState<SessionState>("visible");
+
+  const isInitialBanner = localStorage === "initial-banner";
+
+  useEffect(() => {
+    updateStateValues();
+  }, []);
+
+  useEffect(() => {
+    setIsVisible(!(sessionStorage !== "visible" || localStorage === "none"));
+  }, [localStorage, sessionStorage]);
+
+  const updateStateValues = () => {
+    setLocalStorage(
+      (window.localStorage.getItem("feature-banner") as BannerState) ??
+        "initial-banner"
+    );
+    setSessionStorage(
+      (window.sessionStorage.getItem("feature-banner") as SessionState) ??
+        "visible"
+    );
+  };
```

Comment:
We should write a `useStorage` wrapper for `useState` one day...

Reply:
Yeah this is a great shout, won't do it as part of this PR as it would require QA to re-test again. Will create a follow up ticket for this and this logic can be cleaned up using a `useStorage` hook.